### PR TITLE
refs task #26414 edit question

### PR DESCRIPTION
**Pull request by vuchuc789**
## Related Tickets:
- [#26414](https://edu-redmine.sun-asterisk.vn/issues/26414)

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by vuchuc789** [comment]
ready

**Review by huyenttm-0901 — COMMENTED** [review]
25 files changed, 2 commits

**Comment by vuchuc789** [comment]
ready

File: resources/js/components/EditQuestionForm.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import path from "path";
+
+class CreateQuestionForm {
```
Comment (hoangminhuet) [inline resources/js/components/EditQuestionForm.js]:
```suggestion
class CreateQuestionForm 
{
```

File: resources/js/components/EditQuestionForm.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    constructor(d, bodyEditor) {
```
Comment (hoangminhuet) [inline resources/js/components/EditQuestionForm.js]:
```suggestion
    constructor(d, bodyEditor) 
    {
```

File: resources/js/components/EditQuestionForm.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        this.summitButton = d.getElementById("edit-question-submit");
+        this.bodyEditor = bodyEditor;
+        this.contentId = this.form.getAttribute("content-id").trim();
+
+        this.getContent();
+
+        this.summitButton.addEventListener("click", this.submit.bind(this));
+    }
+
+    submit(e) {
```
Comment (hoangminhuet) [inline resources/js/components/EditQuestionForm.js]:
```suggestion
    submit(e) 
    {
```

File: resources/js/components/EditQuestionForm.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            input.setAttribute("name", "content");
+            input.setAttribute("value", this.bodyEditor.getMarkdown());
+            input.setAttribute("type", "hidden");
+
+            this.form.appendChild(input);
+            this.form.submit();
+        }
+    }
+
+    async getContent() {
```
Comment (hoangminhuet) [inline resources/js/components/EditQuestionForm.js]:
```suggestion
    async getContent() 
    {
```

File: resources/js/components/MarkdownEditor.js [hunk 5]
```diff
@@ -15,6 +15,10 @@ class MarkdownEditor {
     getMarkdown() {
```
Comment (hoangminhuet) [inline resources/js/components/MarkdownEditor.js]:
```suggestion
    getMarkdown() 
    {
```

File: resources/js/components/MarkdownEditor.js [hunk 6]
```diff
@@ -15,6 +15,10 @@ class MarkdownEditor {
...
         return this.editor.getMarkdown();
     }
+
+    setMarkdown(content) {
```
Comment (hoangminhuet) [inline resources/js/components/MarkdownEditor.js]:
```suggestion
    setMarkdown(content) 
    {
```

File: resources/js/components/ShareButtons.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import path from "path";
+
+class ShareButtons {
```
Comment (hoangminhuet) [inline resources/js/components/ShareButtons.js]: tương tự trong file này

File: resources/views/post/edit.blade.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+@section('content')
+    <h2 class="my-5">{{ trans('create.edit') }}</h2>
+    <form method="POST" action="{{ route('questions.update', ['question' => $questionId]) }}"
+        content-id="{{ $content->id }} class=" alert alert-dark" id="edit-question-form">
+        @csrf
+        @method('PUT')
+        <div class="form-group">
+            <label>{{ trans('create.title') }}</label>
+            @error('title')
```
Comment (hoangminhuet) [inline resources/views/post/edit.blade.php]:
```suggestion
            @error ('title')
```

File: resources/views/post/edit.blade.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                <div class="alert alert-danger">{{ $message }}</div>
+            @enderror
+            <small class="form-text text-muted">
+                {{ trans('create.title_description') }}
+            </small>
+            <input class="form-control" type="text" name="title" value="{{ $title }}">
+        </div>
+        <div class="form-group">
+            <label>{{ trans('create.body') }}</label>
+            @error('content')
```
Comment (hoangminhuet) [inline resources/views/post/edit.blade.php]: tương tự

**Comment by vuchuc789** [comment]
ready

**Pull request merged**